### PR TITLE
Replace google.rpc.* with UStatus & UCode

### PR DESCRIPTION
**Pull request by stevenhartley**
The following change will use the ustatus.proto defined in uprotocol-core-api in lieu of googleapis to not have to pull in that complicated dependency and to be in control of our data model.

#45

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -303,7 +291,7 @@
         <dependency>
             <groupId>org.eclipse.uprotocol</groupId>
             <artifactId>uprotocol-core-api</artifactId>
-            <version>1.5.1</version>
+            <version>1.5.2-SNAPSHOT</version>
```
Comment (tamarafischer) [inline pom.xml:294]: why did you bump the version to a SNAPSHOT?

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java [hunk 2]
```diff
@@ -121,16 +122,16 @@ static Optional<String> getToken(CloudEvent cloudEvent) {
     /**
      * Extract the integer value of the communication status attribute from a cloud event. The communication status attribute is optional.
      * If there was a platform communication error that occurred while delivering this cloudEvent, it will be indicated in this attribute.
-     * If the attribute does not exist, it is assumed that everything was Code.OK_VALUE.
+     * If the attribute does not exist, it is assumed that everything was UCode.OK_VALUE.
      * @param cloudEvent CloudEvent with the platformError to be extracted.
-     * @return Returns a {@link Code} value that indicates of a platform communication error while delivering this CloudEvent or Code.OK_VALUE.
+     * @return Returns a {@link Code} value that indicates of a platform communication error while delivering this CloudEvent or UCode.OK_VALUE.
```
Comment (tamarafischer) [inline src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java:127]: The {@Link Code} needs changing as well.

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java [hunk 3]
```diff
@@ -121,16 +122,16 @@ static Optional<String> getToken(CloudEvent cloudEvent) {
...
      */
     static Integer getCommunicationStatus(CloudEvent cloudEvent) {
         try {
             return extractIntegerValueFromExtension("commstatus", cloudEvent)
-                    .orElse(Code.OK_VALUE);
+                    .orElse(UCode.OK_VALUE);
         } catch (Exception e) {
-            return Code.OK_VALUE;
+            return UCode.OK_VALUE;
```
Comment (tamarafischer) [inline src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java:134]: I would expect an exception to be anything but an OK_VALUE.
Is this intended or a mistake

File: src/main/java/org/eclipse/uprotocol/rpc/RpcMapper.java [hunk 4]
```diff
@@ -67,7 +67,7 @@ static <T extends Message> CompletableFuture<T> mapResponse(CompletableFuture<UP
                     return unpackPayload(any, expectedClazz);
                 }
             } catch (InvalidProtocolBufferException e) {
-                throw new RuntimeException(String.format("%s [%s]", e.getMessage(), Status.class.getName()), e);
+                throw new RuntimeException(String.format("%s [%s]", e.getMessage(), UStatus.class.getName()), e);
```
Comment (tamarafischer) [inline src/main/java/org/eclipse/uprotocol/rpc/RpcMapper.java:70]: barfing at the exceptions :)

**Review by tamarafischer — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (stevenhartley) [reply pom.xml:294]: It needs to be snapshot and then the release script will create versions without the SNAPSHOT tag

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java (hunk 2 above)
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java:127]: fixed thanks

File: src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java (hunk 3 above)
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java:134]: fixed thanks

File: src/main/java/org/eclipse/uprotocol/rpc/RpcMapper.java (hunk 4 above)
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/rpc/RpcMapper.java:70]: What should it be?

**Pull request merged**